test/integration/scene_broadcaster_system.cc

```diff
@@ -655,21 +658,61 @@ TEST_P(SceneBroadcasterTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(RemovedComponent))
             return true;
           });
       }
+      // add a component to an entity
```

Comment:
Can we also check changing a component's value and marking it as `PeriodicChange` or `OneTimeChange`? I think this is the trickiest use case to support.

Reply:
Yeah, that's a great point - I added this coverage in https://github.com/ignitionrobotics/ign-gazebo/pull/1392/commits/92cb2f1759d765b9608b526f5ce60e43cb0d4505. In order to process `PeriodicChange` components, [the server has to run unpaused](https://github.com/ignitionrobotics/ign-gazebo/blob/92cb2f1759d765b9608b526f5ce60e43cb0d4505/src/systems/scene_broadcaster/SceneBroadcaster.cc#L342-L343). I found that since we are adding/removing pose components, running the server unpaused results in a crash due to the following code that's part of the physics system update: https://github.com/ignitionrobotics/ign-gazebo/blob/ignition-gazebo6_6.7.0/src/systems/physics/Physics.cc#L2556-L2568

The problem is that physics always expects a pose to exist, but we are removing/re-adding it in this test. So, for the update after removing the pose component, `_ecm.Component<components::Pose>(_model)` gives a `nullptr` that is later dereferenced. I tried to get around this by adding/removing a component that doesn't seem to be used by physics (for example, [`components::Transparency`](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo6/include/ignition/gazebo/components/Transparency.hh)), but I was still getting a segfault. So, I created another test file that uses the `SceneBroadcaster` plugin only and not `Physics`. This test file is used for the particular test that adds/removes components and checks for `SceneBroadcaster` updates (this was done in https://github.com/ignitionrobotics/ign-gazebo/pull/1392/commits/92cb2f1759d765b9608b526f5ce60e43cb0d4505). I'm not sure if there's a better solution than the approach I took, but I spent quite a bit of time trying to debug the segfaults and found that this was the easiest way to avoid this problem for now.

Reply:
As a follow-up, I think that the approach I took in my previous comment of only using `SceneBroadcaster` and not `Physics` for the test is okay since the test is meant to only check update functionality from `SceneBroadcaster`. However, it seems like there may be some places in the physics system where we don't check for potential `nullptr`s because of removed components, and that's the point I was trying to get across in my previous comment. It seems like the segfault issue I was having is outside the scope of this PR, so perhaps we should ticket an issue if this is an actual problem (I think it's probably fair to assume certain components like `Pose` should never be removed from entities, but I wanted to bring this up just to make sure it's not something we have been overlooking).